ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -1319,9 +1319,11 @@ def peer_state_check(self, ip, queue):
         lacp_pdu_times = self.lacp_pdu_times[ip]
         lacp_pdu_down_times = lacp_pdu_times.get("lacp_down")
         lacp_pdu_up_times = lacp_pdu_times.get("lacp_up")
+        self.log('lacp_pdu_down_times: IP:{}: {}'.format(ip, lacp_pdu_down_times))
+        self.log('lacp_pdu_up_times: IP:{}: {}'.format(ip, lacp_pdu_up_times))
         lacp_pdu_before_reboot = float(lacp_pdu_down_times[-1]) if\
             lacp_pdu_down_times and len(lacp_pdu_down_times) > 0 else None
-        lacp_pdu_after_reboot = float(lacp_pdu_up_times[-1]) if\
+        lacp_pdu_after_reboot = float(lacp_pdu_up_times[0]) if\
```

Comment:
Will you get up time from an old up event? Seems that you need to get the first up event of last series of continuous up events?

Reply:
The below check will prevent collecting any UP events before last DOWN.
Therefore we only collect NEW UP events after the last DOWN event.

If there were more than one UP events (which happened in one of the testbeds and triggered this fix), we want the first one (index 0 and not -1).

```Python
 if int(last_lacppdu_time_after_reboot) != int(last_lacppdu_time_before_reboot):
                        self.lacp_pdu_time_on_up.append(last_lacppdu_time_after_reboot)
```